core/src/main/java/com/linecorp/armeria/common/Cookie.java

```diff
@@ -473,6 +473,13 @@ static Cookies fromSetCookieHeaders(boolean strict, Collection<String> setCookie
     @Nullable
     String sameSite();
 
+    /**
+     * Returns whether this {@link Cookie} should only match its original host in domain matching. If this
+     * returns {@code true}, should {@link #domain()} returns a non-null value, it's going to be the original
+     * request host.
+     */
+    boolean isHostOnly();
```

Comment:
Don't we need to make it default to keep binary compatibility just in case? 🧐 (Maybe no users implement this interface)

Reply:
How can it be made default 🤔 ? The host only flag is in the implementation/

Reply:
Yeah. I thought we can make it return `false` because the default value of `hostOnly` in ` CookieBuilder` is `false`.
But it is not a good decision. On second thought, It'd better leave it as it is(abstract method).
Any thoughts @trustin, @minwoox?

Reply:
I'll push other changes first then

Reply:
I guess it's fine because we upgrade the minor version. 🤔